tests/scene/test_curve_3d.h

```diff
@@ -172,9 +172,9 @@ TEST_CASE("[Curve3D] Sampling") {
 	}
 
 	SUBCASE("sample_baked, cubic = true") {
-		CHECK(curve->sample_baked(curve->get_closest_offset(Vector3(0, 0, 0)), true) == Vector3(0, 0, 0));
-		CHECK(curve->sample_baked(curve->get_closest_offset(Vector3(0, 25, 0)), true) == Vector3(0, 25, 0));
-		CHECK(curve->sample_baked(curve->get_closest_offset(Vector3(0, 50, 0)), true) == Vector3(0, 50, 0));
+		CHECK(curve->sample_baked(curve->get_closest_offset(Vector3(0, 0, 0)), true).is_equal_approx(Vector3(0, 0, 0)));
+		CHECK(curve->sample_baked(curve->get_closest_offset(Vector3(0, 25, 0)), true).is_equal_approx(Vector3(0, 25, 0)));
+		CHECK(curve->sample_baked(curve->get_closest_offset(Vector3(0, 50, 0)), true).is_equal_approx(Vector3(0, 50, 0)));
```

Comment:
Does this change mean the new code doesn't provide the exact results? If so I'd call that a regression, no matter the performance gain

Reply:
These are incorrectly done tests. My code may be 0.00000001 different from what it was and only for certain values.
When I tested locally, I didn't see a difference for my data.

Reply:
They aren't if they pass normally, they pass already with the unchanged code so either these tests shouldn't be changed because they pass anyway, or your code changes introduces a loss of precision, the tests are valid if they pass, which they do

Reply:
Maybe I made a bug somewhere on a small calculation.

Reply:
Start by restoring the test and try and make sure your code is passing the tests